marimo/_cli/development/commands.py

```diff
@@ -241,4 +245,77 @@ def openapi() -> None:
     print(yaml.dump(_generate_schema(), default_flow_style=False))
 
 
+@click.group(help="Various commands for the marimo processes", hidden=True)
+def ps() -> None:
+    pass
+
+
+def get_marimo_processes() -> list["psutil.Process"]:
+    import psutil
+
+    def is_marimo_process(proc: psutil.Process) -> bool:
+        if proc.name() == "marimo":
+            return True
+
+        if proc.name().lower() == "python":
+            try:
+                cmds = proc.cmdline()
+            except psutil.AccessDenied:
+                return False
+            except psutil.ZombieProcess:
+                return False
+            # any endswith marimo
+            has_marimo = any(x.endswith("marimo") for x in cmds)
+            # any command equals "tutorial", "edit", or "run"
+            has_running_command = any(
+                x in {"run", "tutorial", "edit"} for x in cmds
+            )
+            return has_marimo and has_running_command
+
+        return False
+
+    result: list[psutil.Process] = []
+
+    for proc in psutil.process_iter():
+        if is_marimo_process(proc):
+            result.append(proc)
+
+    return result
+
+
+@ps.command(help="List the marimo processes", name="list")
+def list_processes() -> None:
```

Comment:
oh interesting - yea we can improve upon this. and maybe write to some output. 

i was just doing this since i was tired of killing my processes. 

it is also under the development commands